superset/viz.py

```diff
@@ -307,7 +307,12 @@ def get_df_payload(self, query_obj=None):
         cached_dttm = datetime.utcnow().isoformat().split('.')[0]
         if cache_key and cache and not self.force:
             cache_value = cache.get(cache_key)
-            if cache_value:
+            perms_decider = config.get('DATA_PERMS_DECIDER')
+            perms_decider_disapproves = (
```

Comment:
`viz.py` doesn't currently handle authorization, I'd rather avoid scattering the auth logic further across modules. Seems like ideally we'd have most security-related logic in `security.py` and have `views/base.py` wrap that up.

Reply:
In our case, we don't want to provide info from the cache based on superset permissions. Rather from an outside system. I moved the check from viz.py to explore_json. and set force=true when the perms_decider denied the person access.